src/Uno.UI/UI/Xaml/Media/Imaging/RenderTargetBitmap.skia.cs

```diff
@@ -0,0 +1,59 @@
+﻿#nullable enable
+using System.Runtime.InteropServices;
+using Windows.Foundation;
+namespace Windows.UI.Xaml.Media.Imaging
+{
+	partial class RenderTargetBitmap
+	{
+		delegate void SwapColor(ref byte[] buffer, int byteCount);
+
+
+		static readonly SwapColor PlatformSwap = RuntimeInformation.IsOSPlatform(OSPlatform.OSX)
+			? new SwapColor(SwapRB)
+			: new SwapColor(DoNothing);
+
+		static void DoNothing(ref byte[] buffer, int byteCount)
+		{
+
+		}
+
+		private static int RendeRenderAsBgra8_Premul(UIElement element, ref byte[]? buffer, Size? scaledSize = null)
+		{
+			var renderSize = element.RenderSize;
+			var visual = element.Visual;
+
+			if (element.RenderSize is { IsEmpty: true }
+			 || element.RenderSize is { Width: 0, Height: 0 })
+			{
+				return 0;
+			}
+
+			var info = new SkiaSharp.
+				SKImageInfo((int)renderSize.Width, (int)renderSize.Height, SkiaSharp.SKColorType.Bgra8888, SkiaSharp.SKAlphaType.Premul);
+			using SkiaSharp.SKSurface surface = SkiaSharp.SKSurface.Create(info);
+			//Ensure Clear
+			var canvas = surface.Canvas;
+			canvas.Clear(SkiaSharp.SKColors.Transparent);
+			visual.Render(surface, info);
+
+			var img = surface.Snapshot();
+
+			var bitmap = SkiaSharp.SKBitmap.FromImage(img);
+			if (scaledSize.HasValue)
+			{
+				var resize = bitmap.Resize(new SkiaSharp.SKImageInfo((int)scaledSize.Value.Width, (int)scaledSize.Value.Height, SkiaSharp.SKColorType.Bgra8888, SkiaSharp.SKAlphaType.Premul)
+					, SkiaSharp.SKFilterQuality.High);
+				bitmap.Dispose();
+				bitmap = resize;
+			}
+
+			var byteCount = bitmap.ByteCount;
+			EnsureBuffer(ref buffer, byteCount);
+			bitmap.GetPixelSpan().CopyTo(buffer);
+			//On macOS color as stored as rgba
+			PlatformSwap(ref buffer!, byteCount);
```

Comment:
I could not identify the source of the problem, it probably depends on the implementation of Skia on macOS, the composition of the pixel and rgba even if set as bgra. I tried with SkPixelMap but the behavior is the same. On Liunx via wsl Skia behaves as expected. Probably need to create a separate test to verify this behavior so that you can quickly identify it should it change.